### ISSUE TITLE
Disable or secure the web interface

### ISSUE THREAD

**Issue by zao** [mumax/3#231]
While testing mumax3/3.10beta when installing it for my cluster users, I observed that it doesn't seem to be possible to disable the web interface nor configure it to use any form of authentication.

Is there any possibility of building mumax3 without the web UI or otherwise disable the automatic enabling of the UI, or at least make it authed?

As our compute nodes and login nodes are globally accessible, the presence of the web UI is an obstacle to providing the software at all on our site.

**Comment by kkingstoun** [comment]
Hi, good question. 

You can try to definitely turn of the web-GUI, maybe by commenting this line goServeGUI() in cmd/mumax/main.go file. 

I'm working on completely new WEB GUI for mumax, base on the VUE.JS and Plotly and Three.JS. I will add token / password authorization on gui, maybe something similar like in Python Jupiter Notebook.

**Comment by godsic** [comment]
@zao Lars, it should be possible to disable web interface by providing an empty '-http=""' flag. Please test and let us know if it works as expected. -mvh

**Comment by zao** [comment]
@godsic Seems like that form indeed disables the web interface, thanks for the fast response.
```
//not starting GUI (-http="")
//openend web interface in x-www-browser
//entering interactive mode
```

The line about trying to open the web interface in a web browser is weird still, but fine as the nodes are mostly headless and won't care.

It doesn't appear when running with an input file specified on the command line, which I'm guessing my users will use.

**Comment by godsic** [comment]
@zao Lars, there is one potential issue still. If users forget to specify *.mx3/*.go script in the slurm job files, then mumax3 process will not terminate, but will be serving WEB interface until terminated from either there or when reaching job time limit. I will fix this ASAP.

**Comment by godsic** [comment]
Fixed in 5cebd32 . `mumax3` won't open interactive session unless `-i` flag is provided.
